File: 00-code(源代码)/src/com/hi/dhl/algorithms/leetcode/_1122/java/Solution.java
package com.hi.dhl.algorithms.leetcode._1122.java;

/**
 * <pre>
 *     author: dhl
 *     date  : 2022/7/31
 *     desc  :
 * </pre>
 */
class Solution {
    public int[] relativeSortArray(int[] arr1, int[] arr2) {
        int[] result = new int[1001];
        int alen = arr1.length;
        for (int i = 0; i < alen; i++) {
            result[arr1[i]]++;
        }

        int index = 0;
        int blen = arr2.length;
        for (int i = 0; i < blen; i++) {
            int key = arr2[i];
            while (result[key] > 0) {
                arr1[index++] = key;
                result[key]--;
            }
        }
        int clen = result.length;
        for (int i = 0; i < clen; i++) {
            while (result[i] > 0) {
                arr1[index++] = i;
                result[i]--;
            }
        }
        return arr1;
    }
}

File: 00-code(源代码)/src/com/hi/dhl/algorithms/leetcode/_1122/kotlin/Solution.kt
package com.hi.dhl.algorithms.leetcode._1122.kotlin

/**
 * <pre>
 *     author: dhl
 *     date  : 2022/7/31
 *     desc  :
 * </pre>
 */
class Solution {
    fun relativeSortArray(arr1: IntArray, arr2: IntArray): IntArray {
        val result = IntArray(1001)
        val alen = arr1.size
        for (i in 0 until alen) {
            result[arr1[i]]++
        }

        var index = 0
        val blen = arr2.size
        for (i in 0 until blen) {
            val key = arr2[i]
            while (result[key] > 0) {
                arr1[index++] = key
                result[key]--
            }
        }

        val clen = result.size
        for (i in 0 until clen) {
            while (result[i] > 0) {
                arr1[index++] = i
                result[i]--
            }
        }
        return arr1
    }
}